src/NVMLDevicePool.hpp

```diff
@@ -50,64 +50,79 @@ namespace geopm
             NVMLDevicePool() = default;
             virtual ~NVMLDevicePool() = default;
             /// @brief Number of accelerators on the platform.
+            /// @return Number of NVML accelerators
             virtual int num_accelerator(void) const = 0;
             /// @brief CPU Affinitization mask for a particular accelerator
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
-            virtual cpu_set_t *ideal_cpu_affinitization_mask(int accel_idx) const = 0;
+            /// @return CPU to Accelerator idea affinitization bitmask
+            virtual cpu_set_t *cpu_affinity_ideal_mask(int accel_idx) const = 0;
             /// @brief Get the NVML device streaming multiprocessor frequency
             //         in MHz
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator streaming multiproccesor frequency in MHz
             virtual uint64_t frequency_status_sm(int accel_idx) const = 0;
             /// @brief Get the NVML device utilization metric
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
-            virtual double utilization(int accel_idx) const = 0;
+            /// @return Accelerator streaming multiprocessor utilization
+            virtual uint64_t utilization(int accel_idx) const = 0;
             /// @brief Get the NVML device power in milliwatts
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator power consumption in milliwatts
             virtual uint64_t power (int accel_idx) const = 0;
             /// @brief Get the NVML device memory subsystem frequency in MHz
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator memory frequency in MHz
             virtual uint64_t frequency_status_mem(int accel_idx) const = 0;
             /// @brief Get the current NVML device clock throttle reasons
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator clock throttle reasons
```

Comment:
Is there an "clock throttle reasons" enum somewhere that you can reference in the documentation?

Reply:
These are currently a series of defines in the NVML API
https://docs.nvidia.com/deploy/nvml-api/group__nvmlClocksThrottleReasons.html#group__nvmlClocksThrottleReasons
There is not currently an enum in GEOPM for this, but we can point to nvml.h